### PR TITLE
#10874: enable initial line allgather testing in TG frequent pipelines

### PR DESCRIPTION
**Pull request by SeanNijjar**
### Ticket
#10874 

### Problem description
We don't have dedicated CCL (in this case line all-gather) testing for TG in our CI pipelines. This PR adds some initial tests to the frequent pipeline.

### What's changed
Added several line all-gather tests to TG frequent pipeline

### Checklist
- [x] Post commit CI passes: https://github.com/tenstorrent/tt-metal/actions/runs/10380932125
- [x] tg frequent: https://github.com/tenstorrent/tt-metal/actions/runs/10422967162
- [x] tg unit: https://github.com/tenstorrent/tt-metal/actions/runs/10390337203
- [x] Blackhole Post commit (if applicable)
- [x] Model regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

Note that the frequent pipeline is currently timing out due to several failing tests. All tests are passing locally:
image

### CODE REVIEW

**Comment by SeanNijjar** [comment]
@tt-rkim is there an alternative place I can put this test list in the meantime?

File: ttnn/cpp/ttnn/operations/ccl/line_all_gather/device/line_all_gather_op.cpp [hunk 1]
```diff
@@ -90,6 +90,30 @@ Tensor line_all_gather(
             const std::vector<std::optional<const Tensor>>& optional_input_tensors,
             const std::vector<std::optional<Tensor>>& optional_output_tensors) mutable -> std::vector<Tensor> {
 
+            std::cout << "YIKES YIKES YIKES YIKES YIKES YIKES YIKES YIKES" << std::endl;
```
Comment (cfjchu) [inline ttnn/cpp/ttnn/operations/ccl/line_all_gather/device/line_all_gather_op.cpp:93]: YIKES, let's remove this 😂

File: ttnn/cpp/ttnn/operations/ccl/line_all_gather/device/line_all_gather_op.cpp [hunk 2]
```diff
@@ -133,6 +157,9 @@ Tensor line_all_gather(
     const auto mesh_view = device_mesh.get_view();
     std::size_t num_devices = (cluster_axis == 0) ? mesh_view->num_rows() : mesh_view->num_cols();
 
+    log_info(tt::LogOp, "dim={}, tensor_shape = [{},{},{},{}]", dim, input_tensor.get_shape()[0], input_tensor.get_shape()[1], input_tensor.get_shape()[2], input_tensor.get_shape()[3]);
```
Comment (cfjchu) [inline ttnn/cpp/ttnn/operations/ccl/line_all_gather/device/line_all_gather_op.cpp:160]: remove

**Review by cfjchu — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/ccl/line_all_gather/device/line_all_gather_op.cpp (hunk 1 above)
Comment (SeanNijjar) [reply ttnn/cpp/ttnn/operations/ccl/line_all_gather/device/line_all_gather_op.cpp:93]: Yea sorry the changes were good to go then I added more changes to enable a new mode and do some debug. I'll pull this back into draft while I do the general cleanup

**Review by ttmchiou — APPROVED** [review]

**Pull request merged**